src/pages/Auth/components/AuthForm/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { Button, Divider, InputAdornment, Stack, TextField } from '@mui/material';
+import { FC, useState, useRef, ChangeEvent, FormEvent } from 'react';
+import { useLoginMutation } from '../../../../api/authApi';
+import { useAppDispatch } from '../../../../store';
+import { setUser } from '../../../../store/slices/currentUserSlice';
+import { getResponseErrorMessage } from '../../../../utils/getResponseErrorMessage';
+import VisibilityIcon from '../../../../components/icons/VisibilityIcon';
+import CustomAlertError from '../../../../components/UI/CustomAlertError';
+import MicrosoftIcon from '../../../../components/icons/MicrosoftIcon';
+
+interface AuthFormProps {}
+
+const AuthForm: FC<AuthFormProps> = ({}) => {
+    const [login, { error }] = useLoginMutation();
```

Comment:
Need to split the logic
- [ ] create a custom hook and move all logic there
- [ ] then, just use this hook (with methods and props needed) in your component